webpages/api/volunteer/volunteer_shift_model.php

```diff
@@ -0,0 +1,243 @@
+            if (mysqli_stmt_execute($stmt)) {
+                mysqli_stmt_close($stmt);
+            } else {
+                throw new DatabaseSqlException("Insert could not be executed: $query");
+            }
+        } else {
+            $query = <<<EOD
+            UPDATE volunteer_shift
+               SET volunteer_job_id = ?, from_time = ?, to_time = ?, location = ?, min_volunteer_count = ?, max_volunteer_count = ?
+             WHERE id = ?;
+            EOD;
+            
+            $stmt = mysqli_prepare($db, $query);
+            mysqli_stmt_bind_param($stmt, "isssiii", $volunteerShift->job->id, $fromTime, $toTime, 
+                $volunteerShift->location, $volunteerShift->minPeople, $volunteerShift->maxPeople, $volunteerShift->id);
+            if (mysqli_stmt_execute($stmt)) {
+                mysqli_stmt_close($stmt);
+            } else {
+                throw new DatabaseSqlException("Insert could not be executed: $query");
+            }
+        }
+    }
+
+    public static function exists($db, $shiftId) {
+        $query = <<<EOD
+        SELECT
+                count(*) as c
+            FROM
+                volunteer_shift S 
+           WHERE id = ?;
+        EOD;
+        
+        $stmt = mysqli_prepare($db, $query);
+        mysqli_stmt_bind_param($stmt, "i", $shiftId);
+        $exists = 0;
+        if (mysqli_stmt_execute($stmt)) {
+            $result = mysqli_stmt_get_result($stmt);
+            while ($row = mysqli_fetch_object($result)) {
+                $exists = $row->c;
+            }
+            mysqli_stmt_close($stmt);
+            return $exists > 0;
+        } else {
+            throw new DatabaseSqlException("Select count(*) command could not be executed: $query");
+        }
+    }
+
+
+    public static function deleteAssignment($db, $badgeId, $shiftId) {
+        $query = <<<EOD
+        DELETE FROM participant_has_volunteer_shift WHERE badgeid = ? and volunteer_shift_id = ?;
+        EOD;
+        
+        $stmt = mysqli_prepare($db, $query);
+        mysqli_stmt_bind_param($stmt, "si", $badgeId, $shiftId);
+        if (mysqli_stmt_execute($stmt)) {
+            mysqli_stmt_close($stmt);
+        } else {
+            throw new DatabaseSqlException("Delete command could not be executed: $query");
+        }
+    }
+
+    public static function deleteShift($db, $shiftId) {
+        $query = <<<EOD
+        DELETE FROM volunteer_shift WHERE id = ?;
+        EOD;
+        
+        $stmt = mysqli_prepare($db, $query);
+        mysqli_stmt_bind_param($stmt, "i", $shiftId);
+        if (mysqli_stmt_execute($stmt)) {
+            mysqli_stmt_close($stmt);
+        } else {
+            throw new DatabaseSqlException("Delete command could not be executed: $query");
+        }
+    }
+
+    public static function createAssignment($db, $badgeId, $shiftId) {
+        $query = <<<EOD
+        INSERT INTO participant_has_volunteer_shift (badgeid, volunteer_shift_id) values (?, ?);
+        EOD;
```

Comment:
EOD.

Reply:
Resolved in 5640d804b653c68d6a5e0897584e38cd4de4808f